Add glow to supporter promo on changelog

jacobvm04 opened the pull request:
Fixes #14303 

before and after:

Website:
![image](https://user-images.githubusercontent.com/35976702/129464437-c9ce7345-cbdc-46d3-8d26-cfbe6b537a96.png)

Lazer before:
![image](https://user-images.githubusercontent.com/35976702/129464441-482e3c8e-3b02-4f75-a677-ce27a9fa2d0d.png)

Lazer after:
![image](https://user-images.githubusercontent.com/35976702/129464450-80a80539-202a-47dd-a420-e6491bdb4b73.png)

I'll note that while the current lazer image is a bit different from the current website one, this is the closest I can get it using EdgeEffect Shadow (instead of CSS's box-shadow, the main difference is that box-shadow has a spread-radius parameter) and the values from the website, but overall they look fairly close.

jacobvm04 committed:
Add glow to supporter promo on changelog

Added glow to the supporter promo at the end of the changelog, as it is on the website.

frenzibyte commented on osu.Game/Overlays/Changelog/ChangelogSupporterPromo.cs:
```diff
@@ -138,14 +138,28 @@ private void load(OsuColour colour, TextureStore textures)
-                    Texture = textures.Get(@"Online/supporter-heart"),
+                    Margin = new MarginPadding { Top = 83 },
+                    Masking = true,
+                    EdgeEffect = new EdgeEffectParameters
+                    {
+                        Type = EdgeEffectType.Shadow,
+                        Colour = Color4.HotPink.Opacity(0.88f),
+                        Offset = new Vector2(0, 0),
+                        Radius = 17,
+                        Roundness = 39f,
+                    },
```
Where did you get these numbers from? they should probably match how web does it if feasible.

jacobvm04 replied:
For the glow shadow, I found that I wouldn't be able to perfectly replicate it because the EdgeEffect Shadow and the CSS box-shadow are implemented differently, so it couldn't be perfectly relicated without writing an EdgeEffect that replicates CSS's box-shadow.

As for the positioning, I found that pulling the numbers straight from the site's code doesn't really work, (As shown in the promo previously being too far up in lazer when the 70px margin from the site was used). I found that increasing that margin put it more in line with the position of the promo as seen on the site.

gagahpangeran replied:
I think it's fine to change the `Radius` to `10` to match osu-web value.

Also the `Roundness` should be `75 / 2` (container width / 2) to make it perfect circle. You can extract the `75` value as const and reuse it for both heart container and heart sprite size to make those two consistent.

gagahpangeran commented on osu.Game/Overlays/Changelog/ChangelogSupporterPromo.cs:
```diff
@@ -138,14 +138,28 @@ private void load(OsuColour colour, TextureStore textures)
                     Height = 75,
-                    Margin = new MarginPadding { Top = 70 },
-                    Texture = textures.Get(@"Online/supporter-heart"),
+                    Margin = new MarginPadding { Top = 83 },
```
Don't change the margin top of the heart.

Move the pippi background slightly above (-28px) to match osu-web. (I was not aware of this when positioning the background).

![Screenshot from 2021-08-15 07-22-31](https://user-images.githubusercontent.com/4964985/129463266-ececc1fc-3789-4d04-9558-51307e0a9e41.png)

gagahpangeran commented on osu.Game/Overlays/Changelog/ChangelogSupporterPromo.cs:
```diff
@@ -138,14 +138,28 @@ private void load(OsuColour colour, TextureStore textures)
+                    EdgeEffect = new EdgeEffectParameters
+                    {
+                        Type = EdgeEffectType.Shadow,
+                        Colour = Color4.HotPink.Opacity(0.88f),
```
This should be `OsuColour.Pink` without any opacity.

gagahpangeran commented on osu.Game/Overlays/Changelog/ChangelogSupporterPromo.cs:
```diff
@@ -138,14 +138,28 @@ private void load(OsuColour colour, TextureStore textures)
+                    {
+                        Type = EdgeEffectType.Shadow,
+                        Colour = Color4.HotPink.Opacity(0.88f),
+                        Offset = new Vector2(0, 0),
```
Can be removed if it just zero.

jacobvm04 committed:
Removed empty offset and corrected colour

jacobvm04 committed:
Corrected pippi background and promo positioning

jacobvm04 commented:
Current before and after:

Website:
![image](https://user-images.githubusercontent.com/35976702/129463937-82ea756d-cb67-4809-b01b-bf131384044d.png)

Lazer before:
![image](https://user-images.githubusercontent.com/35976702/129463941-c374e698-4a87-4eae-b0b4-dfd8f1a815bf.png)

Lazer after:
![image](https://user-images.githubusercontent.com/35976702/129463952-9d51c52e-fdeb-4514-9e36-849227b2f60b.png)

jacobvm04 committed:
Added heart_size constant and adjusted the glow radius value

Added heart_size constant and adjusted the glow radius value to be more in line with the website

jacobvm04 commented:
Current before and after:

Website:
![image](https://user-images.githubusercontent.com/35976702/129464437-c9ce7345-cbdc-46d3-8d26-cfbe6b537a96.png)

Lazer before:
![image](https://user-images.githubusercontent.com/35976702/129464441-482e3c8e-3b02-4f75-a677-ce27a9fa2d0d.png)

Lazer after:
![image](https://user-images.githubusercontent.com/35976702/129464450-80a80539-202a-47dd-a420-e6491bdb4b73.png)

I'll note that while the current lazer image is a bit different from the current website one, this is the closest I can get it using EdgeEffect Shadow and the values from the website, but overall they look fairly close.

frenzibyte commented:
Please update your PR description rather than reposting them every change, that causes noise.

peppy commented:
Has the glow been improved since the OP? It should definitely be possible to almost match web using opacity and radius changes.

bdach commented:
This is how this branch looks as is currently:

![2021-08-15-140441_286x290_scrot](https://user-images.githubusercontent.com/20418176/129477948-bcaa1575-701d-46bb-8e98-55b3060105e1.png)

The glow is still noticeably stronger in-game compared to web, and it seems a little thinner too. For the former, I'm not sure why that's the case (may be color space stuff again?), but the latter part may be due to web specifying a 3px `spread-radius` which doesn't seem to have been accounted for in any capacity yet here. But this is probably just splitting hairs at this point.

jacobvm04 commented:
> Has the glow been improved since the OP? It should definitely be possible to almost match web using opacity and radius changes.

I had actually tweaked the radius and opacity values previously to make it look more similar, however I was recommended to just use the site values instead. For reference, here's how it could look with tweaked values:
Non-tweaked:
![image](https://user-images.githubusercontent.com/35976702/129480267-c9b3befa-3708-4b54-b2fa-d450b043819a.png)

Tweaked (Set color opacity to 0.87 and the EdgeEffect radius to 19):
![image](https://user-images.githubusercontent.com/35976702/129480346-08abebf1-f5e1-4d87-873d-1293e67ba7bc.png)

Website:
![image](https://user-images.githubusercontent.com/35976702/129480324-95b228f5-2b42-438c-af6b-aefe892433d7.png)



> This is how this branch looks as is currently:
> 
> ![2021-08-15-140441_286x290_scrot](https://user-images.githubusercontent.com/20418176/129477948-bcaa1575-701d-46bb-8e98-55b3060105e1.png)
>
> The glow is still noticeably stronger in-game compared to web, and it seems a little thinner too. For the former, I'm not sure why that's the case (may be color space stuff again?), but the latter part may be due to web specifying a 3px `spread-radius` which doesn't seem to have been accounted for in any capacity yet here. But this is probably just splitting hairs at this point.

The key difference from what I've seen seems to be the spread-radius that's in the CSS's box-shadow.

peppy committed:
Merge branch 'master' into add-supporter-glow-changelog

peppy committed:
Simplify size specifications

peppy committed:
Merge branch 'master' into add-supporter-glow-changelog
